### PR TITLE
feat(metrics): RHICOMPL-2056 count policies by OS major version

### PR DESCRIPTION
**Pull request by skateman**
## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [x] Input Validation
- [x] Output Encoding
- [x] Authentication and Password Management
- [x] Session Management
- [x] Access Control
- [x] Cryptographic Practices
- [x] Error Handling and Logging
- [x] Data Protection
- [x] Communication Security
- [x] System Configuration
- [x] Database Security
- [x] File Management
- [x] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: lib/prometheus/business_collector.rb [hunk 1]
```diff
@@ -38,6 +38,13 @@ def initialize
     @client_policies = PrometheusExporter::Metric::Gauge.new(
       'client_policies', 'Policies from clients (excludes Red Hat)'
     )
+    @total_policies_by_os = PrometheusExporter::Metric::Gauge.new(
+      'total_policies_by_os', 'Policies by OS major version'
```
Comment (vkrizan) [inline lib/prometheus/business_collector.rb:42]: Maybe append `_major`. wdyt?

**Review by vkrizan — APPROVED** [review]
LGTM. Thanks!

File: lib/prometheus/business_collector.rb [hunk 2]
```diff
@@ -83,6 +90,15 @@ def collect
           .where(account: client_accounts.select(:account_number)).count
     )
 
+    Policy.joins(:benchmarks).distinct.group('benchmarks.ref_id').count.each do |ref_id, cnt|
+      @total_policies_by_os.observe(cnt, version: ref_id[/(?<=RHEL-)\d/])
```
[the remark this answers is not in the document]
Comment (skateman) [reply lib/prometheus/business_collector.rb:94]: Can do a search and replace in the query if that makes you feel better :beer: :wink:

[the remark this answers is not in the document]
Comment (skateman) [reply lib/prometheus/business_collector.rb:94]: I can try a search and replace in the query

File: lib/prometheus/business_collector.rb (hunk 1 above)
Comment (skateman) [reply lib/prometheus/business_collector.rb:42]: Sure

File: lib/prometheus/business_collector.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (skateman) [reply lib/prometheus/business_collector.rb:94]: Well, it doesn't seem pretty :sweat_smile: 
```ruby
Policy.joins(:benchmarks).distinct.group("REGEXP_REPLACE(benchmarks.ref_id, '^.*RHEL-(\\d)$', '\\1')").count
```
Also we're using the `ref_id[/(?<=RHEL-)\d/]` form elsewhere, so I'm going to keep it consistent...

**Pull request merged**